cartoframes/data/enrichment/polygons_enrichment.py

```diff
@@ -1,83 +1,51 @@
-import uuid
-
-from ..clients import bigquery_client
-from ...auth import get_default_credentials
-from .enrichment_utils import copy_data_and_generate_enrichment_id, process_filters, get_tables_and_variables
-
-
-_ENRICHMENT_ID = 'enrichment_id'
-_WORKING_PROJECT = 'carto-do-customers'
+from .enrichment_service import enrich
 
 # TODO: process column name in metadata, remove spaces and points
 
 
 def enrich_polygons(data, variables, agg_operators, data_geom_column='geometry', filters=dict(), credentials=None):
 
-    credentials = credentials or get_default_credentials()
-    bq_client = bigquery_client.BigQueryClient(credentials)
-
-    user_dataset = credentials.username.replace('-', '_')
-
-    data_copy = copy_data_and_generate_enrichment_id(data, _ENRICHMENT_ID, data_geom_column)
-
-    data_geometry_id_copy = data_copy[[data_geom_column, _ENRICHMENT_ID]]
-    schema = {data_geom_column: 'GEOGRAPHY', _ENRICHMENT_ID: 'INTEGER'}
-
-    id_tablename = uuid.uuid4().hex
-    data_tablename = 'temp_{id}'.format(id=id_tablename)
-
-    bq_client.upload_dataframe(data_geometry_id_copy, schema, data_tablename,
-                               project=_WORKING_PROJECT, dataset=user_dataset, ttl_days=1)
-
-    table_data_enrichment, table_geo_enrichment, variables_list = get_tables_and_variables(variables)
-
-    filters_str = process_filters(filters)
-
-    sql = __prepare_sql(_ENRICHMENT_ID, variables_list, agg_operators, table_data_enrichment,
-                        table_geo_enrichment, user_dataset, _WORKING_PROJECT,
-                        user_dataset, data_tablename, data_geom_column, filters_str)
-
-    data_geometry_id_enriched = bq_client.query(sql).to_dataframe()
-
-    data_copy = data_copy.merge(data_geometry_id_enriched, on=_ENRICHMENT_ID, how='left')\
-        .drop(_ENRICHMENT_ID, axis=1)
+    data_enriched = enrich(_prepare_sql, data=data, variables=variables, agg_operators=agg_operators,
+                           data_geom_column=data_geom_column, filters=filters, credentials=credentials)
```

Comment:
We can avoid requesting this parameter `data_geom_column='geom'` using https://github.com/CartoDB/cartoframes/blob/develop/cartoframes/utils/columns.py#L155. We would need to extract that method first.

Probably it is not a priority

Reply:
What about if the user has two columns with geometries? and what about if the column with geometries have a name not related to usual geometry names?

Reply:
I vote for not supporting dataframes with more than one geometry column by design, otherwise we'll end up trying to do magic to support corner cases.

About geometry column name, it has to be abstracted somehow (the columns module could work) otherwise we'll end up with different logic in different places to do the same thing.

My two cents :)

Reply:
>  vote for not supporting dataframes with more than one geometry column by design

+1 We are doing that way in DB too.

>  we'll end up with different logic in different places to do the same thing

@alejandrohall and I have talked about other things here, mainly describing the initial source of pain is not having geoDataFrames, but this is a bigger task. 

So, I agree with @alrocar, that is the point. We should solve it in a similar fashion in every place. If you want @alejandrohall, I can do it

Reply:
Buy maybe we can do it in a different PR

Reply:
Okey lets remove the geo column name provided by the user!!

Reply:
Moved to an external issue https://github.com/CartoDB/cartoframes/issues/1046